src/main/java/processor/Processor.java

```diff
@@ -0,0 +1,221 @@
+        this.chatbotUi = chatbotUi;
+        //this.file = new File("src/data/tasks.txt");
+        this.storage = new Storage("src/data/tasks.txt", taskList);
+    }
+
+    /**
+     * Processes the given user command by delegating it to the appropriate method in the Processor class.
+     * Specifically, this method processes the user command to mark or unmark a task in the TaskList.
+     * @param userInput the user command to be processed
+     * @throws IOException if an I/O error occurs while processing the command
+     */
+    public void userInputProcessMarkUnmark(String userInput) throws IOException{
+        String[] array = userInput.split(" ");
+        String command = array[0];
+
+        try {
+            int number = Integer.parseInt(array[1]);
+
+            if (command.equals("mark")) {
+                taskList.markTask(number - 1);
+                System.out.println(chatbotUi.dividerWrapper(chatbotUi.mark() + "\n" + taskList.getTaskAtIndex(number - 1)));
+            } else if (command.equals("unmark")) {
+                taskList.unmarkTask(number - 1);
+                System.out.println(chatbotUi.dividerWrapper(chatbotUi.unmark() + "\n" + taskList.getTaskAtIndex(number - 1)));
+            }
+            storage.writeToFile();
+        } catch (NumberFormatException e) {
+            System.out.println(chatbotUi.dividerWrapper("You must use a number to mark."));
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println(chatbotUi.dividerWrapper("You must select a number within the size of the Task List."));
+        }
+    }
+
+
+    /**
+     * Processes the given user command by delegating it to the appropriate method in the Processor class.
+     * Specifically, this method processes the user command to add a task to the TaskList.
+     * @param userInput the user command to be processed
+     * @throws IOException if an I/O error occurs while processing the command
+     */
+    public void userInputAddTask(String userInput) throws IOException {
+        String[] tokens = userInput.split(" ", 2);
+
+        if (tokens.length == 0) {
+            System.out.println(chatbotUi.dividerWrapper("Can not type a blank input!"));
+            return;
+        }
+
+        String command = tokens[0];
+
+        if (!(command.equals("todo") || command.equals("deadline") || command.equals("event"))) {
+            System.out.println(chatbotUi.dividerWrapper("I do not know what type of task that is!"));
+            return;
+        }
+
+       if (tokens.length == 1) {
+           System.out.println(chatbotUi.dividerWrapper("Description of todo can't be blank!"));
+           return;
+       }
+
+        String description = tokens[1].trim();
+
+        String times[] = userInput.split("//");
+
+
+        String desc = times[0];
+        int firstSpaceIndex = desc.indexOf(' ');
+        String substringAfterSpace = desc.substring(firstSpaceIndex + 1);
+
+        switch (command) {
+            case "todo":
```

Comment:
There might be a need to reconfigure your IDE to have no indentations for switch cases to better align with coding standards.

Reply:
Yep I agree. I resolved that later I believe.